typed-racket-test/succeed/prefab.rkt

```diff
@@ -53,3 +53,45 @@
 ;; Test match (indirectly tests unsafe-struct-ref)
 (match (foo 'x) [(foo s) s])
 (match (foo* 'x) [(foo* s) s])
+
+
+
+(struct msg ([str : String] [maybe-str : (Option String)]) #:prefab)
+
+;; tests let bindings fields of prefab structs
+;; works as expected (requires path-type.rkt to be correct w.r.t. prefabs)
+(: test-let-binding (-> Any String))
+(define (test-let-binding x)
+  (cond
+    [(msg? x)
```

Comment:
This doesn't sound like an issue with this PR... it sounds like a fundamental issue with `prefab` structs. For example, this code works in v6.8:

```racket
#lang typed/racket/base
 
(struct fubar ([id : Byte]) #:prefab)
 
(define rval : Any (read (open-input-string "#s(fubar NotANumber)")))
 
 
(when (fubar? rval)
  (displayln (+ 1 (fubar-id rval))))
```

And produces nondeterministic, silly output.